sunraster/tests/test_spectrogram.py

```diff
@@ -61,96 +62,105 @@
-    SOURCE_DATA_DN,
-    wcs=WCS0,
-    extra_coords=EXTRA_COORDS0,
-    unit=u.ct,
-    uncertainty=SOURCE_UNCERTAINTY_DN,
+    SOURCE_DATA_DN, wcs=WCS0, unit=u.ct, uncertainty=SOURCE_UNCERTAINTY_DN, meta=meta_exposure0
 )
+spectrogram_DN0.extra_coords.add(*EXTRA_COORDS0[0])
 spectrogram_DN_per_s0 = SpectrogramCube(
     SOURCE_DATA_DN / SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS0,
     unit=u.ct / u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN / SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_per_s0.extra_coords.add(*EXTRA_COORDS0[0])
 spectrogram_DN_per_s_per_s0 = SpectrogramCube(
     SOURCE_DATA_DN / SINGLES_EXPOSURE_TIME / SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS0,
     unit=u.ct / u.s / u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN / SINGLES_EXPOSURE_TIME / SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_per_s_per_s0.extra_coords.add(*EXTRA_COORDS0[0])
 spectrogram_DN_s0 = SpectrogramCube(
     SOURCE_DATA_DN * SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS0,
     unit=u.ct * u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN * SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_s0.extra_coords.add(*EXTRA_COORDS0[0])
 spectrogram_DN1 = SpectrogramCube(
-    SOURCE_DATA_DN,
-    wcs=WCS0,
-    extra_coords=EXTRA_COORDS1,
-    unit=u.ct,
-    uncertainty=SOURCE_UNCERTAINTY_DN,
+    SOURCE_DATA_DN, wcs=WCS0, unit=u.ct, uncertainty=SOURCE_UNCERTAINTY_DN, meta=meta_exposure0
 )
+spectrogram_DN1.extra_coords.add(*EXTRA_COORDS1[0])
 spectrogram_DN_per_s1 = SpectrogramCube(
     SOURCE_DATA_DN / SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS1,
     unit=u.ct / u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN / SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_per_s1.extra_coords.add(*EXTRA_COORDS1[0])
 spectrogram_DN_per_s_per_s1 = SpectrogramCube(
     SOURCE_DATA_DN / SINGLES_EXPOSURE_TIME / SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS1,
     unit=u.ct / u.s / u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN / SINGLES_EXPOSURE_TIME / SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_per_s_per_s1.extra_coords.add(*EXTRA_COORDS1[0])
 spectrogram_DN_s1 = SpectrogramCube(
     SOURCE_DATA_DN * SINGLES_EXPOSURE_TIME,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS1,
     unit=u.ct * u.s,
     uncertainty=SOURCE_UNCERTAINTY_DN * SINGLES_EXPOSURE_TIME,
+    meta=meta_exposure0,
 )
+spectrogram_DN_s1.extra_coords.add(*EXTRA_COORDS1[0])
 spectrogram_NO_COORDS = SpectrogramCube(SOURCE_DATA_DN, WCS_NO_COORDS)
 spectrogram_instrument_axes = SpectrogramCube(
     SOURCE_DATA_DN,
     wcs=WCS0,
-    extra_coords=EXTRA_COORDS0,
     unit=u.ct,
     uncertainty=SOURCE_UNCERTAINTY_DN,
     mask=MASK,
     instrument_axes=("a", "b", "c"),
+    meta=meta_exposure0,
 )
+spectrogram_instrument_axes.extra_coords.add(*EXTRA_COORDS0[0])
+
+
+def test_str():
```

Comment:
What is this test for?

Reply:
I was trying to debug the str and thought it would be useful to just check that it works without error on our test objects.

Reply:
Do you think it should remain?

Reply:
We should be testing the str methods somewhere since they are quite custom. 

This test tho is bad.